lib/common_fw/common_fw.cpp

```diff
@@ -68,47 +73,99 @@ String make_packet(int id, bool error) {
   incrementPacketCounter();
   return packet;
 }
+//
+// bool parseCommand(String packet) {
+//   Actuator *tmp = decode_received_packet(packet);
+//   if(tmp != nullptr){
+//
+//   }
+// }
 
 /*
  * Decodes a packet sent from ground station in the following format:
  * {<valve_ID>,<open(1) or close(0)|checksum}
  * Populated the fields of the valve and returns the action to be taken
  * This is a pretty beefy function; can we split this up
  */
-int decode_received_packet(String packet, valveInfo *valve, valveInfo valves[], int numValves) {
-  Serial.println(packet);
+int8_t parseCommand(String packet) {
+  debug("Parse Command");
+  debug(packet);
   int data_start_index = packet.indexOf(',');
   if(data_start_index == -1) {
     return -1;
   }
-  int valve_id = packet.substring(1,data_start_index).toInt();
+  debug(String(data_start_index));
+  int actuator_id = packet.substring(1,data_start_index).toInt();
   const int data_end_index = packet.indexOf('|');
   if(data_end_index == -1) {
     return -1;
   }
-  int action = packet.substring(data_start_index + 1,data_end_index).toInt();
+  debug(String(data_end_index));
+  debug(packet.substring(data_start_index + 1,data_end_index).c_str());
+  char data_string[data_end_index - data_start_index + 1];
+  packet.substring(data_start_index + 1,data_end_index).toCharArray(data_string, data_end_index - data_start_index + 1);
+  char *tmp;
+  uint8_t i = 0;
+  float command_data[7];
```

Comment:
should have the maximum # of floats in a command as a constant/#define somewhere else